sfdoc/publish/amazon.py

```diff
@@ -66,26 +67,29 @@ def iter_objects(self, prefix=None):
             else:
                 break
 
-    def process_image(self, filename, bundle):
+    def process_image(self, filename, bundle, rootpath):
         """Upload image file to S3 if needed."""
-        basename = os.path.basename(filename)
-        key = settings.AWS_S3_DRAFT_DIR + basename
+        relative_filename = utils.bundle_relative_path(rootpath, filename)
+        key = settings.AWS_S3_DRAFT_IMG_DIR + relative_filename
         with TemporaryDirectory(prefix=f"process_image_{os.path.basename(filename)}") as tempdir:
-            s3localname = os.path.join(tempdir, basename)
+            s3localname = os.path.join(tempdir, relative_filename)
+            os.makedirs(os.path.dirname(s3localname))
```

Comment:
I haven't tried it yet and don't know whether it would help in your specific uses here, but since we're now using Python 3 you could consider using pathlib for path manipulations.

Reply:
I'll do that next time I need to do path manipulations.